tests/test_server.py

```diff
@@ -187,281 +212,155 @@ def _run_pip(cmd):
     return proc.returncode
 
 
-def _run_pip_install(cmd, port, install_dir, user=None, pswd=None):
-    url = _build_url(port, user, pswd)
-    return _run_pip(f"-vv download -d {install_dir} -i {url} {cmd}")
-
-
 @pytest.fixture
-def pipdir(tmpdir):
-    return tmpdir.mkdir("pip")
+def pipdir(tmp_path_factory):
+    return tmp_path_factory.mktemp("pip")
 
 
 @contextlib.contextmanager
-def pypirc_tmpfile(port, user, password):
-    """Create a temporary pypirc file."""
-    fd, filepath = tempfile.mkstemp()
-    os.close(fd)
-    Path(filepath).write_text(
+def pypirc_file(repo, username="''", password="''"):
+    pypirc_path = Path.home() / ".pypirc"
+    old_pypirc = pypirc_path.read_text() if pypirc_path.is_file() else None
+    pypirc_path.write_text(
         "\n".join(
             (
                 "[distutils]",
                 "index-servers: test",
-                "" "[test]",
-                f"repository: {_build_url(port)}",
-                f"username: {user}",
+                "",
+                "[test]",
+                f"repository: {repo}",
+                f"username: {username}",
                 f"password: {password}",
             )
         )
     )
-
-    print(Path(filepath).read_text())
-    yield filepath
-    os.remove(filepath)
-
-
-@contextlib.contextmanager
-def pypirc_file(txt):
-    pypirc_path = path.local("~/.pypirc", expanduser=1)
-    old_pypirc = pypirc_path.read() if pypirc_path.check() else None
-    pypirc_path.write(txt)
     try:
-        yield
+        yield pypirc_path
     finally:
         if old_pypirc:
-            pypirc_path.write(old_pypirc)
+            pypirc_path.write_text(old_pypirc)
         else:
-            pypirc_path.remove()
+            pypirc_path.unlink()
```

Comment:
At some point we should see if we can figure out a workaround for needing to (even if temporarily) overwrite the pypirc file of whoever's running tests. Not critical at the moment, since it's been that way for ages, but it would be nice

Reply:
Yeah I looked at that. With twine we can specify a specific location for the `.pypirc` file, but for `setup.py` we can't. I can think of a few options though:
- we could change the $HOME when running `setup.py` to point to a different location
- uploading packages with `setup.py` is deprecated so we may consider just removing those tests